lib/install/management/inspect.go

```diff
@@ -89,10 +102,81 @@ func (d *Dispatcher) InspectVCH(vch *vm.VirtualMachine, conf *config.VirtualCont
 		log.Debugf("No host certificates provided")
 	}
 
-	d.ShowVCH(conf, "", "", "", "", "")
+	// Check for valid client cert for a tls-verify configuration
+	if len(conf.CertificateAuthorities) > 0 {
+
+		var possibleCertPaths []string
+		if certPath != "" {
+			log.Infof("cert-path supplied - only checking in %s/ for certs", certPath)
+			possibleCertPaths = append(possibleCertPaths, certPath)
+		} else {
+			possibleCertPaths = append(possibleCertPaths, conf.Name, ".")
+			logMsg := fmt.Sprintf("cert-path not supplied - checking in current directory, %s/", conf.Name)
+
+			dockerConfPath := ""
+			user, err := user.Current()
+			if err == nil {
+				dockerConfPath = filepath.Join(user.HomeDir, ".docker")
+				possibleCertPaths = append(possibleCertPaths, dockerConfPath)
+				logMsg = fmt.Sprintf("%s and %s/", logMsg, dockerConfPath)
+			}
+
+			log.Infof(logMsg)
+		}
+
+		// Check if a valid client cert exists in one of possibleCertPaths
+		certPath = ""
+		for i := range possibleCertPaths {
+			if err = verifyClientCert(conf.CertificateAuthorities, possibleCertPaths[i]); err == nil {
+				certPath = possibleCertPaths[i]
+				break
+			} else {
+				log.Debugf("Unable to verify client cert in %s: %s", possibleCertPaths[i], err)
+			}
+		}
+	}
+
+	d.ShowVCH(conf, "", "", "", "", certPath)
 	return nil
 }
 
+// verifyClientCert checks whether the input path has a client cert that's valid for the VCH.
+func verifyClientCert(ca []byte, path string) error {
```

Comment:
hmm.. we already have similar code in cmd create.go for certificate checking, and that did cname check as well.
In appliance.go, we have another piece of code using certificate to actually connect remote docker service.
I think it would be ideal if refactor create.go and here to use same method, and add a piece of code to connect vch in inspect, with same code used in appliance.go.